### PR TITLE
[IMPROVEMENT] - Project structure and documantation improvements

### PR DESCRIPTION
**Pull request by oguzsh**
## Summary

> [!WARNING]  
> These changes have been tested locally, but double check would be great.

#### Updates:
- Renamed `guncelleme.py` to `update.py` for clarity.
- Updated and refactored `kur.sh` with a new name to improve readability and maintainability.
- Enhanced the README file to provide detailed information about the project and its components.
- Refactored the `felis sh script` and `felis main script` for optimization and better structure.

#### Additions:
- Introduced utility functions to enhance the functionality and flexibility of the project.
- Added validator functions to ensure data integrity and correctness across the application.
- Consolidated all commands in one place for easier access and management.
- Included a `requirements.txt` file for specifying project dependencies.

### CODE REVIEW

File: commands/command.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from enum import Enum
+
+
+class Command(Enum):
```
Comment (oguzsh) [inline commands/command.py:4]: We have a command enum object now. In this way, all commands will be more readable for eng. side.

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    install_python_packages
+
+    SETUP_ROOT="/usr"
+    if [ $(echo $PREFIX | wc -c) -gt 2 ]; then
+        SETUP_ROOT=$PREFIX
+    fi
+
+    mkdir -p "$SETUP_ROOT/share/felis" 2>&1
+    cp felis "$SETUP_ROOT/bin/"
+    cp -r * "$SETUP_ROOT/share/felis/"
```
Comment (oguzsh) [inline setup.sh:50]: We use `cp -r *` instead of `cp *.py` because we have `commands` folder now.

File: felis.py [hunk 3]
```diff
@@ -1,148 +1,47 @@
...
-    os.system("python3 "+prefix+"/share/felis/mp.py -mp4 "+sys.argv[2])
 
-elif sys.argv[1] == "-git":
-    soru=input("\nGüncellemeler Kontrol Edilsin Mi? E/H: ")
-    if soru == "e" or soru == "E":
-        os.system("python3 "+prefix+"/share/felis/guncelleme.py")
-    else:
-        pass
-    os.system("python3 "+prefix+"/share/felis/githubmodul.py")
+class Felis:
```
Comment (oguzsh) [inline felis.py:10]: Converted to class because of MVC principles

File: felis.py [hunk 4]
```diff
@@ -1,148 +1,47 @@
...
+        if command == Command.CHECK_UPDATES:
+            Validator.check_for_updates(self)
+        elif command == Command.REMOVE:
+            remove.call(self=self)
+        elif command in [Command.MUSIC, Command.VIDEO]:
+            media.call(self=self, args=args)
+        elif command in [Command.GIT, Command.REPO]:
+            repo.call(self=self, command=command)
+        elif command == Command.CLONE:
+            clone.call(args=args)
```
Comment (oguzsh) [inline felis.py:36]: Simple command caller structure

**Comment by yilmazemrecom** [comment]
thanks for the big update

**Pull request merged**